Source/ARTStringifiable+Private.h

```diff
@@ -0,0 +1,19 @@
+//
+//  ARTStringifiable+Private.h
+//  Ably
+//
+//  Created by Łukasz Szyszkowski on 21/06/2021.
+//  Copyright © 2021 Ably. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@interface ARTStringifiable()
+
+- (_Nonnull instancetype)initWithString:(nonnull NSString *)value;
+- (_Nonnull instancetype)initWithNumber:(nonnull NSNumber *)value;
+- (_Nonnull instancetype)initWithBool:(BOOL)value;
+
+@property(nonnull, nonatomic, strong) id value;
+
+@end
```

Comment:
```objective-c
// ARTStringifiable.h

+ (nonnull ARTStringifiable*)withString:(nonnull NSString *)value;
+ (nonnull ARTStringifiable*)withNumber:(nonnull NSNumber *)value;
+ (nonnull ARTStringifiable*)withBool:(BOOL)value;
```
only these three static methods can be used to create an instance of this class

```objective-c
// ARTStringifiable+Private.h

- (_Nonnull instancetype)initWithString:(nonnull NSString *)value;
- (_Nonnull instancetype)initWithNumber:(nonnull NSNumber *)value;
- (_Nonnull instancetype)initWithBool:(BOOL)value;
```
these initializers are used internally (not visible to the user). They could be public but I want the user will use only static methods because they are shorter and more concise.

*It's better sometimes to not give a user too many public interfaces ;)*

The `value` property could be defined in `.m` file but since I created `+Private.h` I could out this property there and this is also correct.

Reply:
Okay, I thought they were the same methods (objc is a little hard to read 🥲), but now you've explained they are different. 

> want the user will use only static methods because they are shorter and more concise.

Should the library also use these static methods? Why do we have 2 separate set of functions for public/ private when they do the same thing?

Reply:
I guess using static methods is clear as I explained before.
I need private initializers because I have to initialize somehow this object inside static method implementation.
```objective-c
+ (ARTStringifiable *)withString:(NSString *)value {
    return [[ARTStringifiable alloc] initWithString:value];
}
```

Reply:
Okay I see, we need the initializer to create the object